python_modules/dagster/dagster/_core/definitions/decorators/job_decorator.py

```diff
@@ -140,7 +143,9 @@ def job(
     name: Optional[str] = None,
     description: Optional[str] = None,
     resource_defs: Optional[Mapping[str, ResourceDefinition]] = None,
-    config: Optional[Union[ConfigMapping, Mapping[str, Any], "PartitionedConfig[object]"]] = None,
+    config: Optional[
+        Union[ConfigMapping, Mapping[str, Any], "RunConfig", "PartitionedConfig[object]"]
+    ] = None,
```

Comment:
Minor point here, but I think we should order the types in this Union by probability of desired usage. Meaning I think we want people using RunConfig the most frequently, then the dict, then PartitionedConfig, then ConfigMapping. 

So:
```
config: Optional[
        Union["RunConfig", Mapping[str, Any], "PartitionedConfig[object]", ConfigMapping]
    ] = None,

```